docs/how-to-guides/performing-cluster-maintenance.md

```diff
@@ -0,0 +1,239 @@
+```
+
+## Procedures
+
+### 0. Review the action plan of maintenance mode
+
+The action plan for entering or exiting the maintenance mode can reviewed by using the run dry-run option.
+
+```shell
+juju run microceph/leader exit-maintenance dry-run=True
+juju run microceph/leader enter-maintenance dry-run=True
+```
+
+Some steps in the action plan can be optionally added or removed from the action plan. To see what are the optional
+steps, run
+
+```shell
+juju show-action microceph exit-maintenance
+juju show-action microceph enter-maintenance
+```
+
+### 1. Enter maintenance mode
+
+To put unit `microceph/3` into maintenance mode, and optionally disable the OSD service on that node, run
+
+```shell
+juju run microceph/3 enter-maintenance stop-osds=True
+```
+
+Sample output is:
+
+```text
+Running operation 15 with 1 task
+  - task 16 on unit-microceph-3
+
+Waiting for task 16...
+actions:
+  step-1:
+    description: Check if osds.[4] in node 'juju-681fff-3' are ok-to-stop.
+    error: ""
+    id: check-osd-ok-to-stop-ops
+  step-2:
+    description: Check if there are at least 3 mon, 1 mds, and 1 mgr services in the
+      cluster besides those in node 'juju-681fff-3'
+    error: ""
+    id: check-non-osd-svc-enough-ops
+  step-3:
+    description: Run `ceph osd set noout`.
+    error: ""
+    id: set-noout-ops
+  step-4:
+    description: Assert osd has 'noout' flag set.
+    error: ""
+    id: assert-noout-flag-set-ops
+  step-5:
+    description: Stop osd service in node 'juju-681fff-3'.
+    error: ""
+    id: stop-osd-ops
+errors: ""
+status: success
+```
+
+After entering maintenance mode, the status of the cluster is:
+
+```text
+$ juju ssh microceph/3 -- sudo snap services microceph
+Service               Startup   Current   Notes
+microceph.daemon      enabled   active    -
+microceph.mds         disabled  inactive  -
+microceph.mgr         disabled  inactive  -
+microceph.mon         enabled   active    -
+microceph.osd         disabled  inactive  -
+microceph.rbd-mirror  disabled  inactive  -
+microceph.rgw         disabled  inactive  -
+
+$ juju ssh microceph/3 -- sudo microceph.ceph -s
+  cluster:
+    id:     156f1fa0-51a5-4515-a538-2fe32a8803b5
+    health: HEALTH_WARN
+            noout flag(s) set
+            1 osds down
+            1 host (1 osds) down
+            Degraded data redundancy: 2/6 objects degraded (33.333%), 1 pg degraded, 1 pg undersized
+
+  services:
+    mon: 4 daemons, quorum juju-681fff-1,juju-681fff-0,juju-681fff-2,juju-681fff-3 (age 25m)
+    mgr: juju-681fff-1(active, since 25m), standbys: juju-681fff-0, juju-681fff-2
+    osd: 4 osds: 3 up (since 100s), 4 in (since 24m)
+         flags noout
+
+  data:
+    pools:   1 pools, 1 pgs
+    objects: 2 objects, 449 KiB
+    usage:   107 MiB used, 16 GiB / 16 GiB avail
+    pgs:     2/6 objects degraded (33.333%)
+             1 active+undersized+degraded
+```
+
+For reference, comparing the status of the cluster **before** entering maintenance mode:
```

Comment:
[* reword *]: "Compare the status of the cluster with the cluster status **before** entering maintenance mode: "